addrmgr/netaddress.go

```diff
@@ -113,9 +113,42 @@ func canonicalizeIP(addrType NetAddressType, addrBytes []byte) []byte {
 	return addrBytes
 }
 
-// newNetAddressFromString creates a new address manager network address from the
-// provided string.  The address is expected to be provided in the format
-// host:port.
+// checkNetAddressType returns an error if the suggested address type does not
+// appear to match the provided address.
+func checkNetAddressType(addrType NetAddressType, addrBytes []byte) error {
+	derivedAddressType, err := deriveNetAddressType(addrBytes)
+	if err != nil {
+		return err
+	}
+	if addrType != derivedAddressType {
+		str := fmt.Sprintf("derived address type does not match expected value"+
+			" (got %v, expected %v, address bytes %v).", derivedAddressType,
+			addrType, addrBytes)
+		return makeError(ErrMismatchedAddressType, str)
+	}
+	return nil
+}
+
+// NewNetAddressFromParams creates a new network address from the given
+// parameters. If the provided address type does not appear to match the
+// address, an error is returned.
+func NewNetAddressFromParams(netAddressType NetAddressType, addrBytes []byte, port uint16, timestamp time.Time, services wire.ServiceFlag) (*NetAddress, error) {
```

Comment:
This too is making the same assumption about it being an IP address when it shouldn't be limited to that.